daemon/src/db.rs

```diff
@@ -66,97 +68,83 @@ pub async fn load_order_by_id(
+            oracle_event_id
+
+        from orders
+        where uuid = $1
         "#,
-        uuid
+        id
     )
     .fetch_one(conn)
     .await?;
 
-    let uuid = serde_json::from_str(row.uuid.as_str()).unwrap();
-    let trading_pair = serde_json::from_str(row.trading_pair.as_str()).unwrap();
-    let position = serde_json::from_str(row.position.as_str()).unwrap();
-    let initial_price = serde_json::from_str(row.initial_price.as_str()).unwrap();
-    let min_quantity = serde_json::from_str(row.min_quantity.as_str()).unwrap();
-    let max_quantity = serde_json::from_str(row.max_quantity.as_str()).unwrap();
-    let leverage = Leverage(row.leverage.try_into().unwrap());
-    let liquidation_price = serde_json::from_str(row.liquidation_price.as_str()).unwrap();
-    let creation_timestamp = serde_json::from_str(row.creation_timestamp.as_str()).unwrap();
-    let term = serde_json::from_str(row.term.as_str()).unwrap();
-    let origin = serde_json::from_str(row.origin.as_str()).unwrap();
-
     Ok(Order {
-        id: uuid,
-        trading_pair,
-        position,
-        price: initial_price,
-        min_quantity,
-        max_quantity,
-        leverage,
-        liquidation_price,
-        creation_timestamp,
-        term,
-        origin,
-        oracle_event_id: row.oracle_event_id.parse().unwrap(),
+        id: row.uuid,
+        trading_pair: row.trading_pair,
+        position: row.position,
+        price: Usd(Decimal::from_str(&row.initial_price)?),
+        min_quantity: Usd(Decimal::from_str(&row.min_quantity)?),
+        max_quantity: Usd(Decimal::from_str(&row.max_quantity)?),
+        leverage: row.leverage,
+        liquidation_price: Usd(Decimal::from_str(&row.liquidation_price)?),
+        creation_timestamp: convert_to_system_time(row.ts_secs, row.ts_nanos)?,
+        term: Duration::new(row.term_secs, row.term_nanos),
+        origin: row.origin,
+        oracle_event_id: row.oracle_event_id.parse::<BitMexPriceEventId>()?,
     })
 }
 
 pub async fn insert_cfd(cfd: &Cfd, conn: &mut PoolConnection<Sqlite>) -> anyhow::Result<()> {
-    let mut tx = conn.begin().await?;
-
-    let order_uuid = serde_json::to_string(&cfd.order.id)?;
-    let order_row = sqlx::query!(
+    let order_uuid = &cfd.order.id;
+    let quantity_usd = &cfd.quantity_usd.to_string();
+    let state = serde_json::to_string(&cfd.state)?;
+    sqlx::query!(
         r#"
-        select * from orders where uuid = ?;
-        "#,
-        order_uuid
-    )
-    .fetch_one(&mut tx)
-    .await?;
-
-    let order_id = order_row.id;
-    let quantity_usd = serde_json::to_string(&cfd.quantity_usd)?;
-
-    let cfd_state = serde_json::to_string(&cfd.state)?;
-
-    // save cfd + state in a transaction to make sure the state is only inserted if the cfd was
-    // inserted
+        insert into cfds (
+            order_id,
+            order_uuid,
+            quantity_usd
+        )
+        select
```

Comment:
❓ Why are the selects after the insert needed? Is this here to check that the insert worked - i.e. testing it on the fly?

Interesting pattern - I find it adds some complexity when looking at the coee but don't mind much keeping that.

Reply:
It is in fact [very standard SQL](https://www.w3schools.com/sql/sql_insert_into_select.asp). Basically, you're pointing the output data of the `SELECT` query to be the input data for the `INSERT` query.

Reply:
cool, I did not know that!